lbrynet/daemon/MediaDescriptor.py

```diff
@@ -0,0 +1,19 @@
+import subprocess
+
+class Duration(object):
+    def __init__(self, time):
+        (self.hours, self.minutes, self.seconds) = time
+    def all_in_millisec(self):
+        return (int((float(self.hours) * 3600) +
+               (float(self.minutes) * 60) + (float(self.seconds))) * 1000)
+
+def get_video_length(path):
+    process = subprocess.Popen(['/usr/bin/ffmpeg', '-i', path],
+              stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```

Comment:
this is not a portable way of getting the video duration (won't work on windows, won't work if you have ffmpeg installed elsewhere). 

It also only works with ffmpeg and does not handle cases when the file is some other file type.

Reply:
Did you read my comment? I mentioned it there. I was wondering if someone knows a good portable library that can take duration out of mp4. I also asked on slack. No one has yet pointed one out.

Reply:
Searching around, this looked promising: https://pypi.python.org/pypi/tinytag/